src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs

```diff
@@ -50,39 +59,90 @@ public override async Task InitializeAsync(object navigationData)
+            try
             {
-                Title = "test",
-                Subtitle = "test subtitle",
-                Datetime = "YYYY-MM-DD",
-                Type = TransactionItemType.Actionable.ToString("G")
-            });
-            transactions.Add(new TransactionItem
+                var rsp = await _messageService.SendAsync(context.Wallet, msg, _record, null, true);
+                protocols = rsp.GetMessage<DiscoveryDiscloseMessage>();
+            }
+            catch (Exception) { }
+
+            IList<TransactionItem> transactions = new List<TransactionItem>();
+
+            if (protocols != null)
             {
-                Title = "test",
-                Subtitle = "test subtitle",
-                Datetime = "YYYY-MM-DD",
-                Type = TransactionItemType.Status.ToString("G")
-            });
-            #endif
+                foreach (var protocol in protocols.Protocols)
+                {
+                    switch (protocol.ProtocolId)
+                    {
+                        case "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/trust_ping/1.0":
+                            transactions.Add(new TransactionItem()
+                            {
+                                Title = "Trust Ping",
+                                Subtitle = "Version 1.0",
+                                PrimaryActionTitle = "Ping",
+                                PrimaryActionCommand = new Command(async () =>
+                                {
+                                    await PingConnectionAsync();
+                                }, () => true),
+                                Type = TransactionItemType.Action.ToString("G")
+                            });
+                            break;
+                    }
+                }
+            }
 
             Transactions.Clear();
             Transactions.InsertRange(transactions);
             HasTransactions = transactions.Any();
 
             RefreshingTransactions = false;
-
-            return Task.FromResult(true);
         }
 
+        public async Task PingConnectionAsync()
+        {
+            var dialog = UserDialogs.Instance.Loading("Pinging");
+            var context = await _agentContextProvider.GetContextAsync();
+            var message = new TrustPingMessage
+            {
+                ResponseRequested = true
+            };
+
+            bool success = false;
+            try
+            {
+                var rsp = await _messageService.SendAsync(context.Wallet, message, _record, null, true);
+                var trustPingResponse = rsp.GetMessage<TrustPingResponseMessage>();
+                success = true;
+            }
+            catch (Exception)
+            {
+
+            }
+
+            if (dialog.IsShowing)
+            {
+                dialog.Hide();
+                dialog.Dispose();
+            }
+
+            if (success)
+            {
+                DialogService.Alert("Ping Response Recieved");
```

Comment:
Thoughts on using a ternary operator here?

```suggestion
                DialogService.Alert(
                    success ? 
                    "Ping Response Recieved" :
                    "No Ping Response Recieved"
                );
```